### PR TITLE
Rails Capstone Project

### PR DESCRIPTION
**Pull request by eroiyo**
Hello Microverse community, this is the rails capstone project, the next features have been implemented

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged i

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name
    - amount
    - categories
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

The project has all their components tested, That's all!

The video demonstration is [here](https://www.loom.com/share/229709b8374947abbf57149735059d3e)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,96 @@
...
 
-* System dependencies
+## Built With
 
-* Configuration
+- Ruby on Rails
+- PostgreSQL
 
-* Database creation
+## Live Link
```
Comment (arslanbisharat) [inline README.md:14]: - [ ] In the video recording make sure you add the sentences where you tell us which piece of code you liked the most.

File: db/migrate/20211129201216_add_devise_to_users.rb [hunk 2]
```diff
@@ -22,10 +22,10 @@ def self.up
...
 
       ## Confirmable
-       t.string   :confirmation_token
-       t.datetime :confirmed_at
-       t.datetime :confirmation_sent_at
-       t.string   :unconfirmed_email # Only if using reconfirmable
+       #t.string   :confirmation_token
+       #t.datetime :confirmed_at
+       #t.datetime :confirmation_sent_at
+       #t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (arslanbisharat) [inline db/migrate/20211129201216_add_devise_to_users.rb:28]: - [ ] Let's remove the commented code.

File: spec/features/login_test_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    within('#new_user') do
+      fill_in 'user_email', with: 'test11@example.com'
+      fill_in 'user_password', with: 'taawktljasktlw4aaglj'
+    end
+    click_button 'Log in'
+    User.delete(@user.id)
+    expect(current_path).to eq('/categories')
+  end
+
+  it 'The forms fields are visible' do
```
Comment (arslanbisharat) [inline spec/features/login_test_spec.rb:25]: - [ ] Lets make sure you add more tests cases.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Make sure you follow the correct Gitflow. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: README.md [hunk 4]
```diff
@@ -1,24 +1,96 @@
-# README
+# Budget App
 
-This README would normally document whatever steps are necessary to get the
-application up and running.
+![Microverse](https://img.shields.io/badge/Microverse-blueviolet)
```
Comment (arslanbisharat) [inline README.md:3]: - [ ] You also missed adding Linters.

**Comment by eroiyo** [comment]
Sure, i will be working on it, thanks!

**Comment by eroiyo** [comment]
Link of the video updated, you can click [here ](https://www.loom.com/share/229709b8374947abbf57149735059d3e) also

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @eroiyo 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost-gif
 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by eroiyo** [comment]
Thanks, i will be working on it

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :+1: 

Hello, @eroiyo 👋🏾 👋🏾 congratulations on making it all the project requirements.

Highlights:

- Database is well structured 
- Your project is running successfully
- All tests are passing successfully.

## Optional suggestions

- N/a

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

**Pull request merged**